### PR TITLE
Split out AppBar/SliverAppBar material tests

### PR DESCRIPTION
**Pull request by Piinks**
While looking into resolving https://github.com/flutter/flutter/issues/117903, I found the massive test file `app_bar_test.dart` and found it unwieldy to work with. So before proposing a solution to #117903, which would touch many of these tests, I figured a clean up would be best first.

This splits up `app_bar_test.dart` with a new file `app_bar_sliver_test.dart`, and adds `app_bar_utils.dart` for shared test methods.

It basically moves all SliverAppBar tests into their own file, leaving just AppBar tests in the original file.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/app_bar_utils.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+double tabBarHeight(WidgetTester tester) {
+  return tester.getSize(find.byType(TabBar, skipOffstage: false)).height;
+}
+
+ScrollController primaryScrollController(WidgetTester tester) {
+  return PrimaryScrollController.of(
+    tester.element(find.byType(CustomScrollView))
+  );
+}
+
```
Comment (TahaTesser) [inline packages/flutter/test/material/app_bar_utils.dart:40]: I don't say this often but this class is beautiful!

File: packages/flutter/test/material/app_bar_sliver_test.dart [hunk 2]
```diff
@@ -0,0 +1,2446 @@
...
+    ),
+  );
+}
+
+void main() {
+  setUp(() {
+    debugResetSemanticsIdCounter();
+  });
+
+  group('Material 2', () {
```
Comment (TahaTesser) [inline packages/flutter/test/material/app_bar_sliver_test.dart:58]: Minor nit: Move `Material 2` group to the bottom.

**Review by TahaTesser — APPROVED** [review]
LGTM!

File: packages/flutter/test/material/app_bar_sliver_test.dart (hunk 2 above)
Comment (Piinks) [reply packages/flutter/test/material/app_bar_sliver_test.dart]: Done!

**Comment by Piinks** [comment]
Thanks very much for the review @TahaTesser!

**Pull request merged**